index.js

```diff
@@ -42,17 +43,17 @@ module.exports = function (filename, opts) {
 			return;
 		}
 
-		this.push(new gutil.File({
-			cwd: firstFile.cwd,
-			base: firstFile.base,
-			path: path.join(firstFile.base, filename),
-			contents: zip.generate({
-				type: 'nodebuffer',
-				compression: opts.compress ? 'DEFLATE' : 'STORE',
-				comment: opts.comment
-			})
-		}));
-
-		cb();
+		zip.end(function () {
+			zip.outputStream.pipe(concatStream(function (data) {
```

Comment:
you don't have to wait until the end callback to pipe the output stream. in fact, you can skip the zip.end function entirely - it's just to notify you when it knows the final size of the zip file.